### PR TITLE
[WIP] enable cu116 builds

### PR DESCRIPTION
**Pull request by ptrblck**
Enable and test CUDA11.6 builds

### CODE REVIEW

File: .circleci/docker/build.sh [hunk 1]
```diff
@@ -111,21 +111,9 @@ case "$image" in
     DB=yes
     VISION=yes
     ;;
-  pytorch-linux-xenial-cuda10.2-cudnn7-py3-gcc7)
```
Comment (atalman) [inline .circleci/docker/build.sh:114]: Please original cuda 10.2, cuda 11.3 and cuda 11.5 workflows we need a new one for 11.6

**Comment by atalman** [comment]
We would need to add this change as well: https://github.com/pytorch/pytorch/pull/68665/files

File: .github/scripts/generate_ci_workflows.py [hunk 2]
```diff
@@ -433,6 +433,20 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
             labels={LABEL_CIFLOW_SCHEDULED, LABEL_CIFLOW_CUDA, LABEL_CIFLOW_WIN}
         ),
     ),
+    CIWorkflow(
```
Comment (atalman) [inline .github/scripts/generate_ci_workflows.py:436]: Please don't add windows workflow yet, I need to build an AMI for it

**Comment by ptrblck** [comment]
> We would need to add this change as well: https://github.com/pytorch/pytorch/pull/68665/files

It seems the magma install now uses the `${CUDA_VERSION}` here:
https://github.com/pytorch/pytorch/blob/8a7c9a5e01a24d465126210234aa9d3775b25032/.circleci/docker/common/install_conda.sh#L90
or am I looking at the wrong file?

**Comment by atalman** [comment]
> pytorch/.circleci/docker/common/install_conda.sh

Yes you totally right!

**Comment by atalman** [comment]
@ptrblck Yes please comment out the ``` assert new_periodic == old_periodic ``` in generate_ci_workflows.py and regenerate the workflows.

File: .github/scripts/generate_ci_workflows.py [hunk 3]
```diff
@@ -1265,7 +1288,7 @@ def main() -> None:
     # this is moved to nightly
     old_periodic = old_periodic - {"linux-docs-push"}
 
-    assert new_periodic == old_periodic
+    #assert new_periodic == old_periodic
```
Comment (atalman) [inline .github/scripts/generate_ci_workflows.py:1291]:
```suggestion
    # assert new_periodic == old_periodic
```

**Review by atalman — APPROVED** [review]
LGTM

**Comment by malfet** [comment]
Can we avoid landing PRs with *Work-In-Progress* or *Do not merge* in the title?

**Comment by atalman** [comment]
Sorry about this will remove work-in-progress from title next time when merging

**Pull request closed without merging**